slider/beatmap.py

```diff
@@ -31,6 +31,145 @@ def _get(cs, ix, default=no_default):
         return default
 
 
+class Event:
+
+    EventType = Enum('EventType',
```

Comment:
why not subclass enum? that's how the rest of the enums in slider are defined

Reply:
Afaik subclasses enum doesn't achieve quite what I want.

https://docs.python.org/3/library/enum.html (see restricted subclassing)

I'm (possibly abusing) Enum to allow me to switch on a event type (and construct this event type from the data without needing flags everywhere).

When you subclass Enum, you can't actually define any enums in your base Enum, it's moreso for specifying new behaviour on enums vs having your child classes be members of your enum (if that makes sense).

Reply:
I didn't mean to suggest that `Event` be a subclass of `Enum`, but rather creating a new class `EventType` which is a subclass of enum, instead of having it live as a class attribute here.

Reply:
Hmm, I could pull EventType up one namespace, you reckon that's good enough? I do like the this style syntax a bit more as it's more representative of how I'm using it in a maplike manner.

Reply:
will leave it up to lx10 as this isn't exactly my codebase - just pointing out things I found odd

Reply:
After thinking a bit, decided to go with pulling it out, while it just had data members, I think it was fine as before, but I decided to move event_type parsing into it.